### PR TITLE
Vineet Kumar EquityProductModifications (CDM 6)

### PR DESCRIPTION
**Pull request by regnosys-prod-user**

### CODE REVIEW

File: rosetta-source/src/main/rosetta/product-template-type.rosetta [hunk 1]
```diff
@@ -721,7 +722,7 @@ type ExerciseTerms: <"A class defining the exercise period for an option togethe
...
     earliestExerciseTime BusinessCenterTime (0..1) <"The earliest time at which notice of exercise can be given by the buyer to the seller (or seller's agent) to, and including, the expiration date.">
     latestExerciseTime BusinessCenterTime (0..1) <"For a Bermuda or American style option, the latest time on an exercise business day (excluding the expiration date) within the exercise period that notice can be given by the buyer to the seller or seller's agent. Notice of exercise given after this time will be deemed to have been given on the next exercise business day.">
-    expirationTime BusinessCenterTime (1..1) <"The latest time for exercise on expirationDate. It is made mandatory given that for all option styles, this field is required.">
+    expirationTime BusinessCenterTime (0..1) <"The latest time for exercise on expirationDate. It is made mandatory given that for all option styles, this field is required.">
     expirationTimeType ExpirationTimeTypeEnum (0..1) <"The time of day at which the equity option expires, for example the official closing time of the exchange.">
```
Comment (lolabeis) [inline rosetta-source/src/main/rosetta/product-template-type.rosetta:726]: The idea was to make the `expirationTimeType` attribute mandatory in CDM 6, which we couldn't do in CDM 5. Samples will need to be adjusted accordingly.

File: rosetta-source/src/main/rosetta/product-template-type.rosetta [hunk 2]
```diff
@@ -756,6 +757,11 @@ type ExerciseTerms: <"A class defining the exercise period for an option togethe
                 and earliestExerciseTime exists
                 and partialExercise is absent
                 and exerciseFee is absent
+    
+    condition ExpirationTimeChoice: <"Condition to validate the existence of correlation between expirationTime and expirationTimeType">
+        (if expirationTime exists and expirationTimeType exists then expirationTimeType = ExpirationTimeTypeEnum -> SpecificTime)
+        and 
+        (if expirationTimeType exists and expirationTimeType = ExpirationTimeTypeEnum -> SpecificTime then expirationTime exists)
```
Comment (lolabeis) [inline rosetta-source/src/main/rosetta/product-template-type.rosetta:764]: Condition logic can be simplified given that `expirationTimeType` is mandatory.

File: rosetta-source/src/main/rosetta/product-qualification-func.rosetta [hunk 3]
```diff
@@ -720,7 +721,10 @@ func Qualify_EquityOption_PriceReturnBasicPerformance_SingleName: <"Qualifies a
             and optionUnderlier -> Observable -> Asset -> Instrument -> Security only exists 
                     // qualifies that no feature other than averaging exists
             and (economicTerms -> payout -> optionPayout -> feature is absent
-                or economicTerms -> payout -> optionPayout -> feature -> averagingFeature only exists)
+                or economicTerms -> payout -> optionPayout -> feature -> averagingFeature only exists
+                or economicTerms -> payout -> optionPayout -> feature -> fxFeature only-element -> quanto  exists
+                or economicTerms -> payout -> optionPayout -> feature -> fxFeature only-element -> composite  exists)
```
Comment (lolabeis) [inline rosetta-source/src/main/rosetta/product-qualification-func.rosetta:726]: Why are these 2 additional assertions (for quanto and composite) already there in CDM 5, but not in CDM 6? Are we sure we also need to add them in CDM 6?

File: rosetta-source/src/main/rosetta/product-template-type.rosetta [hunk 4]
```diff
@@ -722,7 +723,7 @@ type ExerciseTerms: <"A class defining the exercise period for an option togethe
     earliestExerciseTime BusinessCenterTime (0..1) <"The earliest time at which notice of exercise can be given by the buyer to the seller (or seller's agent) to, and including, the expiration date.">
     latestExerciseTime BusinessCenterTime (0..1) <"For a Bermuda or American style option, the latest time on an exercise business day (excluding the expiration date) within the exercise period that notice can be given by the buyer to the seller or seller's agent. Notice of exercise given after this time will be deemed to have been given on the next exercise business day.">
     expirationTime BusinessCenterTime (1..1) <"The latest time for exercise on expirationDate. It is made mandatory given that for all option styles, this field is required.">
```
Comment (lolabeis) [inline rosetta-source/src/main/rosetta/product-template-type.rosetta:725]: The `expirationTime` attribute needs to be made optional, and the condition over its absence or existence in relation to `expirationTimeType` be reinstated

File: rosetta-source/src/main/rosetta/product-template-type.rosetta [hunk 5]
```diff
@@ -756,9 +757,13 @@ type ExerciseTerms: <"A class defining the exercise period for an option togethe
                 and earliestExerciseTime exists
                 and partialExercise is absent
                 and exerciseFee is absent
+    
+    condition ExpirationTimeChoice: <"Condition to validate the existence of correlation between expirationTime and expirationTimeType">
```
Comment (lolabeis) [inline rosetta-source/src/main/rosetta/product-template-type.rosetta:761]: Remove "existence of": "Condition to validate the correlation..."

File: rosetta-source/src/main/rosetta/product-template-type.rosetta [hunk 6]
```diff
@@ -756,9 +757,13 @@ type ExerciseTerms: <"A class defining the exercise period for an option togethe
...
+        (if expirationTime exists then expirationTimeType = ExpirationTimeTypeEnum -> SpecificTime or 
+        if expirationTimeType = ExpirationTimeTypeEnum -> SpecificTime then expirationTime exists)
```
Comment (lolabeis) [inline rosetta-source/src/main/rosetta/product-template-type.rosetta:763]: Should be `and` not `or`

File: rosetta-source/src/main/rosetta/product-template-type.rosetta [hunk 7]
```diff
@@ -260,6 +260,7 @@ type Payout: <"Represents the set of future cashflow methodologies in the form o
         if optionPayout exists
         then optionPayout -> priceQuantity exists
                 or optionPayout -> underlier -> Product ->> economicTerms -> payout -> interestRatePayout count = 2
+                or optionPayout -> underlier -> Observable -> Asset -> Cash exists
```
Comment (lolabeis) [inline rosetta-source/src/main/rosetta/product-template-type.rosetta:263]: This last `or` (which was included in CDM 5 to cover the foreign exchange case) becomes unnecessary in CDM 6. Post the asset refactoring work, the quantity should be populated for foreign exchange options.

Comment (winitall01) [reply rosetta-source/src/main/rosetta/product-template-type.rosetta]: @lolabeis - Changes made to the PR

**Comment by hugohills-regnosys** [comment]
Replaced by https://github.com/finos/common-domain-model/pull/3278

**Pull request closed without merging**